samples/vboxwrapper/vbox_vboxmanage.cpp

```diff
@@ -58,821 +58,816 @@ using std::string;
+        if (enable_floppyio) {
+            vboxlog_msg("Removing virtual floppy disk from VirtualBox.");
+            command  = "closemedium floppy \"" + virtual_machine_slot_directory + "/" + floppy_image_filename + "\" ";
             if (delete_media) {
                 command += "--delete ";
             }
 
-            vbm_popen(command, output, "remove virtual cache disk", false, false);
+            vbm_popen(command, output, "remove virtual floppy disk", false, false);
         }
-    } 
-
-    if (enable_floppyio) {
-        vboxlog_msg("Removing virtual floppy disk from VirtualBox.");
-        command  = "closemedium floppy \"" + virtual_machine_slot_directory + "/" + floppy_image_filename + "\" ";
-        if (delete_media) {
-            command += "--delete ";
-        }
-
-        vbm_popen(command, output, "remove virtual floppy disk", false, false);
-    }
-
-    return 0;
-}
-
-int VBOX_VM::deregister_stale_vm() {
-    string command;
-    string output;
-    string virtual_machine_slot_directory;
-    size_t uuid_start;
-    size_t uuid_end;
-    int retval;
-
-    get_slot_directory(virtual_machine_slot_directory);
-
-    // Output from showhdinfo should look a little like this:
-    //   UUID:                 c119acaf-636c-41f6-86c9-38e639a31339
-    //   Accessible:           yes
-    //   Logical size:         10240 MBytes
-    //   Current size on disk: 0 MBytes
-    //   Type:                 normal (base)
-    //   Storage format:       VDI
-    //   Format variant:       dynamic default
-    //   In use by VMs:        test2 (UUID: 000ab2be-1254-4c6a-9fdc-1536a478f601)
-    //   Location:             C:\Users\romw\VirtualBox VMs\test2\test2.vdi
-    //
-    if (enable_isocontextualization) {
-        command  = "showhdinfo \"" + virtual_machine_slot_directory + "/" + cache_disk_filename + "\" ";
-        retval = vbm_popen(command, output, "get HDD info");
-        if (retval) return retval;
+        return 0;
+    }
+
+    int VBOX_VM::deregister_stale_vm() {
+        string command;
+        string output;
+        string virtual_machine_slot_directory;
+        size_t uuid_start;
+        size_t uuid_end;
+        int retval;
+
+        get_slot_directory(virtual_machine_slot_directory);
+
+        // Output from showhdinfo should look a little like this:
+        //   UUID:                 c119acaf-636c-41f6-86c9-38e639a31339
+        //   Accessible:           yes
+        //   Logical size:         10240 MBytes
+        //   Current size on disk: 0 MBytes
+        //   Type:                 normal (base)
+        //   Storage format:       VDI
+        //   Format variant:       dynamic default
+        //   In use by VMs:        test2 (UUID: 000ab2be-1254-4c6a-9fdc-1536a478f601)
+        //   Location:             C:\Users\romw\VirtualBox VMs\test2\test2.vdi
+        //
+        if (enable_isocontextualization && enable_isocontextualization) {
```

Comment:
Testing the same flag twice. It should probably be `if (enable_isocontextualization && enable_cache_disk)` since the code is checking for cache disk and to match [the code in the COM interface](https://github.com/BOINC/boinc/blob/master/samples/vboxwrapper/vbox_mscom_impl.cpp#L1171). Although, the code seem confused:

```
if (enable_isocontextualization && ???) {
    blah
} else {
    if (...) {
        blah
    } else {
        if (enable_isocontextualization) {
            if (enable_cache_disk) {
```

It seems one can't have cache disk without ISO contextualization so the last one or two tests are always false. I'm not sure how it's supposed to work. @romw comment?

Reply:
This code changed a bit since Juhas comment but still needs to be cleaned because it now tests the same flag twice.